app/Http/Controllers/QuestionAttachmentController.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\QuestionAttachment;
+use Illuminate\Http\Request;
+
+class QuestionAttachmentController extends Controller
+{
+    public function show(QuestionAttachment $attachment)
+    {
+        if(auth()->user()->current_level == $attachment->question->level) {
+            return response()->file(\Storage::path($attachment->path));
```

Comment:
import storage facade as `use Illuminate\Support\Facades\Storage;`, avoid using shorthand for importing classes it reduces readability of code. Use Intellephense  extension if you work on vscode and enable format code onsave option.